File: solution/0000-0099/0056.Merge Intervals/Solution.ts
function merge(intervals: number[][]): number[][] {
    intervals.sort((a, b) => a[0] - b[0]);
    const ans: number[][] = [];
    let [start, end] = intervals[0];
    for (const [s, e] of intervals.slice(1)) {
        if (end < s) {
            ans.push([start, end]);
            [start, end] = [s, e];
        } else {
            end = Math.max(end, e);
        }
    }
    ans.push([start, end]);
    return ans;
};

File: solution/0000-0099/0056.Merge Intervals/Solution.go
func merge(intervals [][]int) (ans [][]int) {
    sort.Slice(intervals, func(i, j int) bool {
        return intervals[i][0] < intervals[j][0]
    })
    start, end := intervals[0][0], intervals[0][1]
    for _, e := range intervals[1:] {
        if end < e[0] {
            ans = append(ans, []int{start, end})
            start, end = e[0], e[1]
        } else if end < e[1] {
            end = e[1]
        }
    }
    ans = append(ans, []int{start, end})
    return ans
}
